src/Workspaces/MSBuild/BuildHost/Build/ProjectBuildManager.cs

```diff
@@ -151,6 +166,32 @@ private ImmutableDictionary<string, string> AllGlobalProperties
         }
     }
 
+    public (MSB.Evaluation.Project? project, DiagnosticLog log) LoadProject(string path, Stream readStream)
+    {
+        var log = new DiagnosticLog();
+        try
+        {
+            var projectCollection = new MSB.Evaluation.ProjectCollection(
```

Comment:
I admit I don't know the significance of this not using the batch project collection, or whether this creates problems around the loggers conflicting like you saw. That said, I'm not sure how to answer that question either. If we already had an existing bug there, then perhaps that can be counted as tracking the issue and this is fine.

Reply:
Tracking under https://github.com/dotnet/sdk/issues/48708.